mempool/src/core_mempool/transaction_store.rs

```diff
@@ -153,27 +154,25 @@ impl TransactionStore {
     /// check if transaction is already present in Mempool
     /// e.g. given request is update
     /// we allow increase in gas price to speed up process
-    fn check_for_update(
+    fn handle_gas_price_update(
         &mut self,
         txn: &MempoolTransaction,
-    ) -> (bool, MempoolAddTransactionStatus) {
-        let mut is_update = false;
+    ) -> (MempoolAddTransactionStatus) {
```

Comment:
nit: no need for brackets here

Reply:
this function can instead return Result<()> (error in case of invalidupdate). Then callsite can just do:
```
if self.handle_gas_update(txn).is_err() {
     return MempoolAddTransactionStatusCode::InvalidUpdate;
}
```